Buildings/Fluid/Humidifiers/DXDehumidifier.mo

```diff
@@ -0,0 +1,289 @@
+within Buildings.Fluid.Humidifiers;
+model DXDehumidifier "DX dehumidifier"
+  extends Buildings.Fluid.Interfaces.PartialTwoPort;
+
+  parameter Modelica.Units.SI.VolumeFlowRate V_flow_nominal(min=0) = 5.805556e-7
+    "Rated water removal rate, in m3/s"
+    annotation (Dialog(group="Nominal condition"));
+
+  parameter Boolean addPowerToMedium = true
+    "Transfer power and heat to the fluid medium";
+
+  parameter Real eneFac_nominal(
+    final min=0) = 3.412
+    "Rated energy factor, in L/kwh"
+    annotation (Dialog(group="Nominal condition"));
+
+  parameter Buildings.Fluid.Humidifiers.Data.Generic per
+    "Performance data"
+    annotation (choicesAllMatching=true,
+      Placement(transformation(extent={{60,-56},{80,-36}})));
+
+  parameter Modelica.Units.SI.MassFlowRate m_flow_nominal
+    "Nominal mass flow rate";
+
+  parameter Modelica.Units.SI.PressureDifference dp_nominal
+    "Pressure difference";
+
+  Modelica.Blocks.Interfaces.BooleanInput uEna
+    "Enable signal"
+    annotation (Placement(transformation(extent={{-120,-50},{-100,-30}}),
+      iconTransformation(extent={{-120,-50},{-100,-30}})));
+
+  Modelica.Thermal.HeatTransfer.Sources.PrescribedHeatFlow preHeaFlo
+    "Heat transfer into medium from dehumidifying action"
+    annotation (Placement(transformation(extent={{-50,50},{-30,70}})));
+
+  Modelica.Thermal.HeatTransfer.Sensors.HeatFlowSensor heaFloSen
+    "Sensor for measuring heat flow rate into medium"
+    annotation (Placement(transformation(extent={{-20,50},{0,70}})));
+
+  Modelica.Thermal.HeatTransfer.Sensors.TemperatureSensor senTem
+    "Temperature sensor"
+    annotation (Placement(transformation(extent={{38,50},{58,70}})));
+
+  Modelica.Blocks.Interfaces.RealOutput T(
```

Comment:
best should make sure the connectors are in the same sections of the code. The ideal order would be parameter section, connector sections, and other instances section.